torch/_functorch/aot_autograd.py

```diff
@@ -971,76 +1089,77 @@ def same_dtype_views(view1, view2):
 #   f(a, c.view(-1), b.view(-1), b, c, d)
 # Modified argument list:
 #   c_base comes first because the first c view came earlier in arg list than the first b view
+#   a and d still show up in the modified arg list, but b and c don't- they're regenerated from their bases
 #   b_base = torch.Tensor(b.storage())
 #   c_base = torch.Tensor(c.storage())
 #   f(c_base, b_base, a, d)
 def merge_view_inputs(
     fwd_inputs: List[Any],
-    mutated_input_info: List[MutationType]
-) -> Tuple[List[Any], Optional[List[Union[int, Tuple[int, Tuple[Any]]]]]]:
+    mutated_input_info: List[InputAliasInfo]
+) -> Tuple[List[Any], Optional[List[Union[int, Tuple[int, torch.Tensor]]]]]:
     assert len(fwd_inputs) == len(mutated_input_info)
     storage_ref_to_idx: Dict[StorageWeakRef, List[int]] = collections.defaultdict(list)
+    base_args = []
+    other_args = []
     for i, inpt in enumerate(fwd_inputs):
         if isinstance(inpt, Tensor):
             storage_ref = StorageWeakRef(inpt.storage())
             storage_ref_to_idx[storage_ref].append(i)
-    base_args = []
-    other_args = []
+        else:
+            other_args.append(inpt)
     # This list contains metadata that tells you what the i'th argument in the inner calling convention should be.
     # It's either:
     # - another int (corresponding to the index in the argument list of the element from the outer calling convention)
-    # - idx, *args, where we can generate the new output with old_args[idx].as_strided(*args)
+    # - idx, view_tensor, where we can generate the new output with view_tensor._view_func(old_args[idx])
     #   idx corresponds to which synthetic base from the outer calling context to view
-    inner_calling_convention_meta: Dict[int, Union[int, Tuple[int, List[Any]]]] = {}
+    inner_calling_convention_meta: Dict[int, Union[int, Tuple[int, torch.Tensor]]] = {}
     for aliased_input_indices in storage_ref_to_idx.values():
-        if len(aliased_input_indices) > 1 and any(
+        if len(aliased_input_indices) <= 1 or not any(
```

Comment:
what happened here?

Reply:
this was a small piece of feedback from the original input mutation PR: inverting the condition so the giant block doesn't need indenting